modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -107,14 +112,20 @@ public <T> Either<String, T> getValue() {
         }
 
         return (Either<String, T>) originalValue
-            .mapRight(v -> {
-              if (v instanceof String) {
-                return macroExpandedRawValue((String) v);
-              } else if (getDecorated().getModel().getName().equals(ERROR_MAPPINGS_PARAMETER_NAME)) {
-                return mapErrorMappings(getDecorated());
-              }
-              return v;
-            });
+            .mapRight(this::mapComponent);
+      }
+
+      private <T> Object mapComponent(T rawValue) {
+        if (rawValue instanceof ArrayList) {
+          return rawValue;
```

Comment:
what is this case needed for?

Reply:
[Arrays](https://github.com/mulesoft/mule-artifact-ast/blob/8f3a72a3698b447c0503b0d8f4751bef0043852a/mule-ast/src/main/java/org/mule/runtime/ast/api/util/MuleAstUtils.java#L220) and [Maps](https://github.com/mulesoft/mule-artifact-ast/blob/8f3a72a3698b447c0503b0d8f4751bef0043852a/mule-ast/src/main/java/org/mule/runtime/ast/api/util/MuleAstUtils.java#L230) are contemplated by the code, so when it comes one of them, the rawValue is returned as it is. But with the new change as an Array supports Child Declaration, and doesn't need to be treated as other components, we have to exclude it and keep the same behaviour.